src/ncp-spinel/SpinelNCPControlInterface.cpp

```diff
@@ -429,11 +438,25 @@ SpinelNCPControlInterface::permit_join(
 			SPINEL_PROP_THREAD_ASSISTING_PORTS,
 			ntohs(traffic_port)
 		));
+
+		memcpy(steering_data_addr, mNCPInstance->mSteeringDataAddress, sizeof(steering_data_addr));
+
 	} else {
+
 		factory.add_command(SpinelPackData(
 			SPINEL_FRAME_PACK_CMD_PROP_VALUE_SET(SPINEL_DATATYPE_NULL_S),
 			SPINEL_PROP_THREAD_ASSISTING_PORTS
 		));
+
+		memset(steering_data_addr, 0, sizeof(steering_data_addr));
```

Comment:
`should_update_steering_data` is updated in `if` before this (it checks if the capability is present and we have `mSetSteeringDataWhenJoinable` set). 

Here we only decide what to set it to (either all zero, or the value `mSteeringDataAddress`)...